network/secure_session_wrapper.go

```diff
@@ -180,23 +182,100 @@ func (err *ConnectionWrapError) Temporary() bool {
 // SecureSessionEstablishingTimeout timeout for secure session handshake that should be enough
 const SecureSessionEstablishingTimeout = time.Second * 10
 
+// GRPCAuthType auth type used for acra-connector
+const GRPCAuthType = "acra-connector"
+
 // MaxClientIDDataLength max data length of first packet that send from client. 1 kb was chosen manually and that should
 // be enough
 const MaxClientIDDataLength = 1024 // 1kb
 
 // ErrClientIDPacketToBig show that packet with ClientID too big
 var ErrClientIDPacketToBig = errors.New("packet with ClientID too big")
 
+// SecureSessionInfo struct used to store metadata between connector<->server/translator and implement credentials.AuthInfo [grpc] interface
+// to pass these to end handlers
+type SecureSessionInfo struct {
+	clientID    []byte
+	spanContext trace.SpanContext
+}
+
+// ClientID return cliendID
+func (s SecureSessionInfo) ClientID() []byte {
+	return s.clientID
+}
+
+// SpanContext return context from connector
+func (s SecureSessionInfo) SpanContext() trace.SpanContext {
+	return s.spanContext
+}
+
+// AuthType return acra-connector's AuthType identifier for gRPC transport
+func (s SecureSessionInfo) AuthType() string {
+	return GRPCAuthType
+}
+
 // SecureSessionConnectionWrapper adds SecureSession encryption above connection
 type SecureSessionConnectionWrapper struct {
 	keystore         keystore.SecureSessionKeyStore
 	handshakeTimeout time.Duration
 	id               []byte
+	serverID         []byte
+}
+
+// ClientHandshake wrap outcoming client's connection to server with secure session as gRPC transport
+func (wrapper *SecureSessionConnectionWrapper) ClientHandshake(ctx context.Context, authority string, conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	wrappedConn, _, err := wrapper.wrap(conn, false)
+	if err != nil {
+		log.WithError(err).Errorln("Can't wrap server's connection")
+		return conn, nil, err
+	}
+	return wrappedConn, SecureSessionInfo{clientID: wrapper.id}, nil
+}
+
+// ServerHandshake wrap incoming client's connection with secure session as gRPC transport
+func (wrapper *SecureSessionConnectionWrapper) ServerHandshake(conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	wrappedConn, clientID, err := wrapper.wrap(conn, true)
+	if err != nil {
+		log.WithError(err).Errorln("Can't wrap client's connection")
+		return conn, nil, err
+	}
+	spnCtx, err := ReadTrace(wrappedConn)
+	if err != nil {
+		return wrappedConn, nil, err
+	}
+	return wrappedConn, SecureSessionInfo{clientID: clientID, spanContext: spnCtx}, nil
+}
+
+// Info return protocol info for Secure Session
+func (wrapper *SecureSessionConnectionWrapper) Info() credentials.ProtocolInfo {
+	return credentials.ProtocolInfo{
+		ProtocolVersion:  "1.0.0",
+		SecurityProtocol: "secure session",
+		SecurityVersion:  "1.0.0.",
```

Comment:
Where do we verify these versions? 🤔

Reply:
it was added for `credentials.TransportCredentials` interface (gRPC) to use secure session as transport and authentication backend for gRPC and it require to define some info like this. And I just defined simple values like 1.0.0 with prediction that we will not change them in close future